### ISSUE TITLE
[BUG ?] Paperless-ngx 2.2.1 + Gotenberg 7.10.0 / 7.10.1 / 7.10.2

### ISSUE THREAD

**Issue by ctout76** [gotenberg/gotenberg#763]
Happy New Year!

Hello, I am encountering a problem from Gotenberg version 7.10.0 (same problem with versions 7.10.0 - 7.10.1 - 7.10.2)
For example, when converting .xlsx or .pptx files, I often get the following error message:

file.xlsx: Error occurred while consuming document file.xlsx: Error while converting document to PDF: Server error '503 Service Unavailable' for url 'http://gotenberg:3000/forms/libreoffice/convert'

As soon as I force install Gotenberg 7.9.2, I don't encounter any problems !

**Comment by gulien** [comment]
Hello @ctout76 👋 the LibreOffice versions are not the same between 7.9 & 7.10, so I guess something changed. Did you try to increase the API timeout?

**Comment by ctout76** [comment]
Hello, thank you for the response.

I'm running Synology with Docker and I saw that another user is also having a problem with Docker:

https://github.com/gotenberg/gotenberg/issues/764

Here is my configuration, I tried to add the line "--libreoffice-start-timeout=30s", without improvement for me.

    command:
       - "gotenberg"
       - "--chromium-disable-javascript=true"
       - "--chromium-disable-routes=true"
       - "--chromium-allow-list=file:///tmp/.*"
       - "--chromium-failed-starts-threshold=15"
       - "--log-level=debug"
       - "--api-timeout=120s"
       - "--uno-listener-restart-threshold=30"

**Comment by togut** [comment]
Hello, I'm the user experiencing an issue with Docker. #764
Though our problems seem related, there are distinct differences, so I've created a separate issue for clarity.

I noticed that your log level is set to debug. Please try adding the command "--libreoffice-auto-start=true," as I have in my setup, and then share the log entries?

We may be facing a common underlying issue, with the difference being that, in your case, the error occurs as soon as Gotenberg attempts to start LibreOffice on a conversion request.

**Comment by gulien** [comment]
I wonder if your Gotenberg instances are not resource starved 🤔 For whatever reasons, the new LibreOffice might consume more resources?

**Comment by gulien** [comment]
Also, what are your architectures?

**Comment by togut** [comment]
I believe that the resources are fine.
As I mentioned in #764:

> - Checking system resources and stopping all other containers outside of the PaperlessNGX Configuration
> > Result: There's enough RAM and processor capacity available (at least 512 MB). No change in conditions, as the issue eventually resolves.

I am not sure if this is what you are asking for, but this comes out if I prompt `uname -a`
`Linux ServerName 4.4.302+ #69057 SMP Mon Nov 13 14:21:00 CST 2023 x86_64 GNU/Linux synology_apollolake_918+`

Also, as mentioned in #764 these are my system specifications:
```Model: Synology DS918+
CPU: Intel Celeron J3455 Quad-Core, 1.5 GHz (base)
RAM: 4 GB DDR3L
DSM Version: DSM 7.2.1-69057 Update 3`

**Comment by gulien** [comment]
I have no idea why LibreOffice struggles to startup in your case. Does it always eventually resolve? Randomly (e.g., it could work the first time or after X attempts)?

**Comment by gulien** [comment]
The debug log level is already catching the stdout/stderr of the LibreOffice process, and AFAIK there is no verbose mode.

I am not able to reproduce the issue locally or in the demo instance (1 vCPU / 5120MB RAM), so I don’t know 😕

You’re both using Synology, I guess as NAS?

**Comment by vore** [comment]
Hello! 
I have some issues as well. When I import some .docx file to my Paperless-NGX I get the following error message: `Error while converting document to PDF: Client error '404 Not Found' for url 'http://gotenberg:3000/forms/libreoffice/convert/forms/libreoffice/convert#'` Maybe the same issue? Also using a NAS (QNAP) with 16GB Ram and Intel Celeron J4025 CPU (2 Cores @ 2,9Ghz).

**Comment by gulien** [comment]
@vore your URL is not correct lol not the same

**Comment by togut** [comment]
@vore and @ctout76 could you please set the `"--libreoffice-auto-start=true,"` and `"--log-level=debug"` as configured in #764 within your docker-compose.yml and post the logs?
This could help us analyze if we have the same problem or get further information.
Thank you!

**Comment by ctout76** [comment]
> @vore and @ctout76 could you please set the `"--libreoffice-auto-start=true,"` and `"--log-level=debug"` as configured in #764 within your docker-compose.yml and post the logs? This could help us analyze if we have the same problem or get further information. Thank you!

Hi, OK, I'm carrying out the tests but I'll take a little time to respond [a lot of work :-) ]

**Comment by ctout76** [comment]
Hi, I took advantage of brand new updates to test with Gotenberg 8.0.0 by removing all flags except "--libreoffice-auto-start=true"

So far, I haven't had any errors... Fingers crossed  🤞

Paperless-ngx 2.3.0
PostgreSQL 16.1
REDIS 7.2.3
Gotenberg 8.0.0

  gotenberg:
    image: gotenberg/gotenberg:8.0.0
    restart: on-failure:5
    container_name: PaperlessNGX-GOTENBERG
    ports:
      - 3000:3000
    command:
      - "gotenberg"
      - "--libreoffice-auto-start=true"


I'm continuing the tests !

**Comment by ctout76** [comment]
2024/01/06 15:55:23	stderr	{"level":"error","ts":1704552923.6439111,"logger":"api","msg":"convert to PDF: process restart before task: restart process: start process: LibreOffice socket not available: context deadline exceeded","trace":"49272765-a0c2-4bc3-9868-849dbe68d7e6","remote_ip":"172.22.0.6","host":"gotenberg:3000","uri":"/forms/libreoffice/convert","method":"POST","path":"/forms/libreoffice/convert","referer":"","user_agent":"python-httpx/0.25.2","status":503,"latency":20013621130,"latency_human":"20.01362113s","bytes_in":1826613,"bytes_out":19}

2024/01/06 15:55:02	stderr	{"level":"error","ts":1704552902.543403,"logger":"api","msg":"convert to PDF: process restart before task: restart process: start process: LibreOffice socket not available: context deadline exceeded","trace":"8509d3ef-90d8-4b59-a8d4-cb71f6ef6a24","remote_ip":"172.22.0.6","host":"gotenberg:3000","uri":"/forms/libreoffice/convert","method":"POST","path":"/forms/libreoffice/convert","referer":"","user_agent":"python-httpx/0.25.2","status":503,"latency":20021183283,"latency_human":"20.021183283s","bytes_in":1725937,"bytes_out":19}

2024/01/06 15:49:26	stderr	{"level":"error","ts":1704552566.5785854,"logger":"api","msg":"convert to PDF: process restart before task: restart process: start process: LibreOffice socket not available: context deadline exceeded","trace":"278deea5-535e-41d5-89a6-d8caa44f09a8","remote_ip":"172.22.0.6","host":"gotenberg:3000","uri":"/forms/libreoffice/convert","method":"POST","path":"/forms/libreoffice/convert","referer":"","user_agent":"python-httpx/0.25.2","status":503,"latency":20006224699,"latency_human":"20.006224699s","bytes_in":9890,"bytes_out":19}

**Comment by ctout76** [comment]
Much less error with these settings (Paperless-ngx 2.3.0 - PostgreSQL 16.1 - REDIS 7.2.3)

    command:
      - "gotenberg"
      - "--chromium-disable-javascript=true"
      - "--chromium-disable-routes=true"
      - "--chromium-allow-list=file:///tmp/.*"
      - "--api-timeout=120s"


2024/01/06 17:02:13	stderr	{"level":"error","ts":1704556933.0748959,"logger":"api","msg":"convert to PDF: process first start: start process: LibreOffice socket not available: context deadline exceeded","trace":"4f83335f-7dce-4d0b-a522-1dc84331bfc1","remote_ip":"172.18.0.4","host":"gotenberg:3000","uri":"/forms/libreoffice/convert","method":"POST","path":"/forms/libreoffice/convert","referer":"","user_agent":"python-httpx/0.25.2","status":503,"latency":20016617190,"latency_human":"20.01661719s","bytes_in":150059,"bytes_out":19}
2024/01/06 17:00:48	stdout	[SYSTEM] prometheus: collecting metrics
2024/01/06 17:00:48	stdout	[SYSTEM] pdfengines: libreoffice-pdfengine pdfcpu pdftk qpdf
2024/01/06 17:00:48	stdout	[SYSTEM] api: server listening on port 3000
2024/01/06 17:00:48	stdout	[SYSTEM] chromium: Chromium ready to start
2024/01/06 17:00:48	stdout	[SYSTEM] libreoffice-api: LibreOffice ready to start
2024/01/06 17:00:48	stdout	[SYSTEM] modules: api chromium libreoffice libreoffice-api libreoffice-pdfengine logging pdfcpu pdfengines pdftk prometheus qpdf webhook 
2024/01/06 17:00:48	stdout	-------------------------------------------------------
2024/01/06 17:00:48	stdout	Version: 8.0.0
2024/01/06 17:00:48	stdout	A Docker-powered stateless API for PDF files.
2024/01/06 17:00:48	stdout	
2024/01/06 17:00:48	stdout	                                   /___/
2024/01/06 17:00:48	stdout	\___/\___/\__/\__/_//_/_.__/\__/_/  \_, / 
2024/01/06 17:00:48	stdout	/ (_ / _ \/ __/ -_) _ \/ _ \/ -_) __/ _ '/
2024/01/06 17:00:48	stdout	 / ___/__  / /____ ___  / /  ___ _______ _
2024/01/06 17:00:48	stdout	  _____     __           __               
2024/01/06 17:00:48	stdout	
2024/01/06 16:58:30	stdout	[SYSTEM] api: application stopped
2024/01/06 16:58:30	stdout	[SYSTEM] prometheus: application stopped
2024/01/06 16:58:30	stdout	[SYSTEM] graceful shutdown of 30s
2024/01/06 16:57:51	stdout	[SYSTEM] pdfengines: libreoffice-pdfengine pdfcpu pdftk qpdf
2024/01/06 16:57:51	stdout	[SYSTEM] prometheus: collecting metrics
2024/01/06 16:57:51	stdout	[SYSTEM] api: server listening on port 3000
2024/01/06 16:57:51	stdout	[SYSTEM] chromium: Chromium ready to start
2024/01/06 16:57:51	stdout	[SYSTEM] libreoffice-api: LibreOffice ready to start
2024/01/06 16:57:51	stdout	[SYSTEM] modules: api chromium libreoffice libreoffice-api libreoffice-pdfengine logging pdfcpu pdfengines pdftk prometheus qpdf webhook 
2024/01/06 16:57:51	stdout	-------------------------------------------------------
2024/01/06 16:57:51	stdout	Version: 8.0.0
2024/01/06 16:57:51	stdout	A Docker-powered stateless API for PDF files.
2024/01/06 16:57:51	stdout	
2024/01/06 16:57:51	stdout	                                   /___/
2024/01/06 16:57:51	stdout	\___/\___/\__/\__/_//_/_.__/\__/_/  \_, / 
2024/01/06 16:57:51	stdout	/ (_ / _ \/ __/ -_) _ \/ _ \/ -_) __/ _ '/
2024/01/06 16:57:51	stdout	 / ___/__  / /____ ___  / /  ___ _______ _
2024/01/06 16:57:51	stdout	  _____     __           __               
2024/01/06 16:57:51	stdout

**Comment by gulien** [comment]
@ctout76 thanks! What if you increase the LibreOffice timeout with an aberrant value like `1000s`?

**Comment by ctout76** [comment]
Tests on 19 files (1 failed) + log-level=debug

**Comment by ctout76** [comment]
> @ctout76 thanks! What if you increase the LibreOffice timeout with an aberrant value like `1000s`?

I will test immediately

**Comment by ctout76** [comment]
> > @ctout76 thanks! What if you increase the LibreOffice timeout with an aberrant value like `1000s`?
> 
> I will test immediately

There are still '503 Service Unavailable' issues

```
essai 002AR.xlsx: Error occurred while consuming document essai 002AR.xlsx: Error while converting document to PDF: Server error '503 Service Unavailable' for url 'http://gotenberg:3000/forms/libreoffice/convert'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503
```

**Comment by ctout76** [comment]
I tried with exactly the same files +
Paperless-ngx 2.3.0
PostgreSQL 16.1
REDIS 7.2.3
Gotenberg 7.9.2
     command:
       - "gotenberg"
       - "--chromium-disable-javascript=true"
       - "--chromium-disable-routes=true"
       - "--chromium-allow-list=file:///tmp/.*"
       - "--chromium-failed-starts-threshold=15"
       - "--log-level=debug"
       - "--api-timeout=120s"
       - "--uno-listener-restart-threshold=30"

0 problems

**Comment by gulien** [comment]
IMO the versions 7.6 and 7.5 of LibreOffice are somehow the culprit.

I guess you can mitigate the issue with `--libreoffice-restart-after=30` or stick with version `7.9.2`.

**Comment by togut** [comment]
I could not reproduce the problem outside of my Synology server.
The problem persists but I don't have an approach we could try yet.

Either way, I am planning a RAM upgrade. Hardware should arrive this week. I'll give an update once I've done the upgrade and tested to see if it fixes the problem, though hopes are low.

**Comment by gulien** [comment]
Could be the CPU also 😬

**Comment by OverDodo** [comment]
I am on a Synology DS723+, so x86_64. I also get 503 errors quite randomly. Sometimes on the first document, so when there is 0 load. Feels completely random. This is the debug log for when it happens. Currently on gotenberg 8.0.1. My system has 18GB of RAM, which it doesnt even get to 50%. Increased the api timeout to 300s, but the error occurrs much quicker. 

`
DBG ts=1705074998.8316917 logger=libreoffice-api.libreoffice msg=process is unhealthy, cannot handle task, restarting...
DBG ts=1705074998.8317118 logger=libreoffice-api.libreoffice msg=restart process
DBG ts=1705074998.8317175 logger=libreoffice-api.libreoffice msg=shutdown process
DBG ts=1705074998.8317192 logger=api.formslibreofficeconvert msg='/tmp/48ac6250-4ab2-4fdf-93ca-5cd6a440ccff/14943012-52f0-4367-ae27-be93fbd203ef' context's working directory removed trace=08c60e09-8e40-4162-ba7a-1efc90f27fae
ERR ts=1705074998.831743 logger=api msg=convert to PDF: process restart before task: restart process: start process: LibreOffice socket not available: context deadline exceeded trace=08c60e09-8e40-4162-ba7a-1efc90f27fae remote_ip=172.20.200.10 host=testgotenberg:3000 uri=/forms/libreoffice/convert method=POST path=/forms/libreoffice/convert referer= user_agent=python-httpx/0.26.0 status=503 latency=39828987229 latency_human=39.828987229s bytes_in=57481 bytes_out=19
`

**Comment by ctout76** [comment]
After more than 30 hours of testing with 600 .doc and .xls files, I offer you this configuration !
Tested on 3 different machines.

My default configuration:

Synology DS920+  8G
Paperless-ngx 2.5.0
Postgres 16.1
Redis 7.2.4
Gotenbeg 8.0.3

Don't forget to personalize:  UID and GID - TZ - IP adress

In the DSM configuration: Enable Reverse Proxy + Websocket, - Enable HTTP Compression - Enable port reuse to improve the capacity of simultaneous connections for the web service

Reboot the machine


EDIT: https://github.com/gotenberg/gotenberg/issues/763#issuecomment-2790238910

```
version: "3.9"
services:
  redis:
    image: redis:7.2.4
    command:
      - /bin/sh
      - -c
      - redis-server --requirepass redispass
    container_name: PaperlessNGX-REDIS
    hostname: paper-redis
    mem_limit: 512m
    mem_reservation: 256m
    cpu_shares: 768
    security_opt:
      - no-new-privileges:true
    read_only: true
    user: 1026:100
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping || exit 1"]
    volumes:
      - /volume1/docker/paperlessngx/redis:/data:rw
    environment:
      TZ: Europe/Paris
    restart: on-failure:5

  db:
    image: postgres:16.1
    container_name: PaperlessNGX-DB
    hostname: paper-db
    mem_limit: 1g
    cpu_shares: 768
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "paperless", "-U", "paperlessuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/paperlessngx/db:/var/lib/postgresql/data:rw
    environment:
      POSTGRES_DB: paperless
      POSTGRES_USER: paperlessuser
      POSTGRES_PASSWORD: paperlesspass
    restart: on-failure:5

  gotenberg:
    image: gotenberg/gotenberg:8.0.3
    container_name: PaperlessNGX-GOTENBERG
    hostname: gotenberg
    mem_limit: 4g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    user: 1026:100
    command:
      - "gotenberg"
      - "--chromium-disable-javascript=true"
      - "--chromium-disable-routes=false"
      - "--chromium-allow-list=file:///tmp/.*"
      - "--api-timeout=200s"
      - "--libreoffice-auto-start=true"
      - "--libreoffice-restart-after=0"
    restart: on-failure:5

  tika:
    image: ghcr.io/paperless-ngx/tika:latest
    container_name: PaperlessNGX-TIKA
    hostname: tika
    security_opt:
      - no-new-privileges:true
    user: 1026:100
    restart: on-failure:5

  paperless:
    image: ghcr.io/paperless-ngx/paperless-ngx:2.5.0
    container_name: PaperlessNGX
    hostname: paperless-ngx
    mem_limit: 4g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "curl", "-fs", "-S", "--max-time", "2", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    ports:
      - 8777:8000
    volumes:
      - /volume1/docker/paperlessngx/data:/usr/src/paperless/data:rw
      - /volume1/docker/paperlessngx/media:/usr/src/paperless/media:rw
      - /volume1/docker/paperlessngx/export:/usr/src/paperless/export:rw
      - /volume1/docker/paperlessngx/consume:/usr/src/paperless/consume:rw
    environment:
      PAPERLESS_REDIS: redis://:redispass@paper-redis:6379
      PAPERLESS_DBENGINE: postgresql
      PAPERLESS_DBHOST: paper-db
      PAPERLESS_DBNAME: paperless
      PAPERLESS_DBUSER: paperlessuser
      PAPERLESS_DBPASS: paperlesspass
      PAPERLESS_WEBSERVER_WORKERS: 2
      PAPERLESS_TASK_WORKERS: 1
      USERMAP_UID: 1026
      USERMAP_GID: 100
      PAPERLESS_ENABLE_COMPRESSION: 0
      PAPERLESS_TIME_ZONE: Europe/Paris
      PAPERLESS_ADMIN_USER: admin
      PAPERLESS_ADMIN_PASSWORD: paperless
      PAPERLESS_URL: https://XXX.XXX.X.XX:XXXX
      PAPERLESS_CSRF_TRUSTED_ORIGINS: https://XXX.XXX.X.XX:XXXX
      PAPERLESS_OCR_LANGUAGE: fra+eng
      PAPERLESS_OCR_USER_ARGS: '{"invalidate_digital_signatures": true}'
      PAPERLESS_TIKA_ENABLED: 1
      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
      PAPERLESS_TIKA_ENDPOINT: http://tika:9998
    restart: on-failure:5
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_healthy
      tika:
        condition: service_started
      gotenberg:
        condition: service_started

```
Please let me know if this works for you !

**Comment by OverDodo** [comment]
> After more than 30 hours of testing with 600 .doc and .xls files, I offer you this configuration ! Tested on 3 different machines.
> 
> My default configuration:
> 
> Synology DS920+ 8G Paperless-ngx 2.3.3 Postgres 16.1 Redis 7.2.4 Gotenbeg 8.0.1
> 
> Don't forget to personalize: UID and GID - TZ - IP adress
> 
> In the DSM configuration: Enable Reverse Proxy + Websocket, - Enable HTTP Compression - Enable port reuse to improve the capacity of simultaneous connections for the web service
> 
> Reboot the machine
> 
> ```
> version: "3.9"
> services:
>   redis:
>     image: redis:7.2.4
>     command:
>       - /bin/sh
>       - -c
>       - redis-server --requirepass redispass
>     container_name: PaperlessNGX-REDIS
>     hostname: paper-redis
>     mem_limit: 512m
>     mem_reservation: 256m
>     cpu_shares: 768
>     security_opt:
>       - no-new-privileges:true
>     read_only: true
>     user: 1026:100
>     healthcheck:
>       test: ["CMD-SHELL", "redis-cli ping || exit 1"]
>     volumes:
>       - /volume1/docker/paperlessngx/redis:/data:rw
>     environment:
>       TZ: Europe/Paris
>     restart: on-failure:5
> 
>   db:
>     image: postgres:16.1
>     container_name: PaperlessNGX-DB
>     hostname: paper-db
>     mem_limit: 1g
>     cpu_shares: 768
>     security_opt:
>       - no-new-privileges:true
>     healthcheck:
>       test: ["CMD", "pg_isready", "-q", "-d", "paperless", "-U", "paperlessuser"]
>       timeout: 45s
>       interval: 10s
>       retries: 10
>     volumes:
>       - /volume1/docker/paperlessngx/db:/var/lib/postgresql/data:rw
>     environment:
>       POSTGRES_DB: paperless
>       POSTGRES_USER: paperlessuser
>       POSTGRES_PASSWORD: paperlesspass
>     restart: on-failure:5
> 
>   gotenberg:
>     image: gotenberg/gotenberg:8.0.1
>     container_name: PaperlessNGX-GOTENBERG
>     hostname: gotenberg
>     mem_limit: 4g
>     cpu_shares: 1024
>     security_opt:
>       - no-new-privileges:true
>     user: 1026:100
>     command:
>       - "gotenberg"
>       - "--chromium-disable-javascript=true"
>       - "--chromium-disable-routes=true"
>       - "--chromium-allow-list=file:///tmp/.*"
>       - "--api-timeout=200s"
>       - "--libreoffice-auto-start=true"
>       - "--libreoffice-restart-after=0"
>     restart: on-failure:5
> 
>   tika:
>     image: ghcr.io/paperless-ngx/tika:latest
>     container_name: PaperlessNGX-TIKA
>     hostname: tika
>     security_opt:
>       - no-new-privileges:true
>     user: 1026:100
>     restart: on-failure:5
> 
>   paperless:
>     image: ghcr.io/paperless-ngx/paperless-ngx:2.3.3
>     container_name: PaperlessNGX
>     hostname: paperless-ngx
>     mem_limit: 4g
>     cpu_shares: 1024
>     security_opt:
>       - no-new-privileges:true
>     healthcheck:
>       test: ["CMD", "curl", "-fs", "-S", "--max-time", "2", "http://localhost:8000"]
>       interval: 30s
>       timeout: 10s
>       retries: 5
>     ports:
>       - 8777:8000
>     volumes:
>       - /volume1/docker/paperlessngx/data:/usr/src/paperless/data:rw
>       - /volume1/docker/paperlessngx/media:/usr/src/paperless/media:rw
>       - /volume1/docker/paperlessngx/export:/usr/src/paperless/export:rw
>       - /volume1/docker/paperlessngx/consume:/usr/src/paperless/consume:rw
>     environment:
>       PAPERLESS_REDIS: redis://:redispass@paper-redis:6379
>       PAPERLESS_DBENGINE: postgresql
>       PAPERLESS_DBHOST: paper-db
>       PAPERLESS_DBNAME: paperless
>       PAPERLESS_DBUSER: paperlessuser
>       PAPERLESS_DBPASS: paperlesspass
>       PAPERLESS_TASK_WORKERS: 1
>       USERMAP_UID: 1026
>       USERMAP_GID: 100
>       PAPERLESS_ENABLE_COMPRESSION: 0
>       PAPERLESS_TIME_ZONE: Europe/Paris
>       PAPERLESS_ADMIN_USER: admin
>       PAPERLESS_ADMIN_PASSWORD: paperless
>       PAPERLESS_URL: https://XXX.XXX.X.XX:XXXX
>       PAPERLESS_CSRF_TRUSTED_ORIGINS: https://XXX.XXX.X.XX:XXXX
>       PAPERLESS_OCR_LANGUAGE: fra+eng
>       PAPERLESS_OCR_USER_ARGS: '{"invalidate_digital_signatures": true}'
>       PAPERLESS_TIKA_ENABLED: 1
>       PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
>       PAPERLESS_TIKA_ENDPOINT: http://tika:9998
>     restart: on-failure:5
>     depends_on:
>       db:
>         condition: service_healthy
>       redis:
>         condition: service_healthy
>       tika:
>         condition: service_started
>       gotenberg:
>         condition: service_started
> ```
> 
> Please let me know if this works for you !

Thanks alot for the compose! I just tried throwing 200 documents at it, no issues at all! Works perfectly! Quick question, since you have a very similar setup. Does the mail convert work for you? I always get a 404 Not Found error, when it tries to convert an email to pdf..

Error while converting email to PDF: Client error '404 Not Found' for url 'http://testgotenberg:3000/forms/chromium/convert/html'

**Comment by ctout76** [comment]
> > 
> 
> Thanks alot for the compose! I just tried throwing 200 documents at it, no issues at all! Works perfectly! Quick question, since you have a very similar setup. Does the mail convert work for you? I always get a 404 Not Found error, when it tries to convert an email to pdf.. I thought I'd ask here before opening another issue that might be user dependent..
> 
> Error while converting email to PDF: Client error '404 Not Found' for url 'http://testgotenberg:3000/forms/chromium/convert/html'

Super !

No, email conversion does not work, this will be the next problem to resolve...

**Comment by gulien** [comment]
@ctout76 thanks a lot for taking the time to investigate!

**Comment by ctout76** [comment]
EDIT: Editing line in my Docker Compose -> Add `PAPERLESS_WEBSERVER_WORKERS: 2`

**Comment by ctout76** [comment]
EDIT 2: Editing line in the Docker Compose -> deleting `"--chromium-disable-routes=true"` (mail converter)

**Comment by ctout76** [comment]
> > > 
> > 
> > 
> > Thanks alot for the compose! I just tried throwing 200 documents at it, no issues at all! Works perfectly! Quick question, since you have a very similar setup. Does the mail convert work for you? I always get a 404 Not Found error, when it tries to convert an email to pdf.. I thought I'd ask here before opening another issue that might be user dependent..
> > Error while converting email to PDF: Client error '404 Not Found' for url 'http://testgotenberg:3000/forms/chromium/convert/html'
> 
> Super !
> 
> No, email conversion does not work, this will be the next problem to resolve...

EDIT 2: Editing line in the Docker Compose -> "--chromium-disable-routes=false" (enable mail converter)

🤦 email conversion will work 😄

**Comment by OverDodo** [comment]
> > > > 
> > > 
> > > 
> > > Thanks alot for the compose! I just tried throwing 200 documents at it, no issues at all! Works perfectly! Quick question, since you have a very similar setup. Does the mail convert work for you? I always get a 404 Not Found error, when it tries to convert an email to pdf.. I thought I'd ask here before opening another issue that might be user dependent..
> > > Error while converting email to PDF: Client error '404 Not Found' for url 'http://testgotenberg:3000/forms/chromium/convert/html'
> > 
> > 
> > Super !
> > No, email conversion does not work, this will be the next problem to resolve...
> 
> EDIT 2: Editing line in the Docker Compose -> "--chromium-disable-routes=false" (enable mail converter)
> 
> 🤦 email conversion will work 😄

Awesome, just tried it and it worked flawlessly! Thank you very much 😊

**Comment by ctout76** [comment]
EDIT 3: Editing lines in the Docker Compose -> image: gotenberg/gotenberg:8.0.3 + image: ghcr.io/paperless-ngx/paperless-ngx:2.5.0
=> OK

**Comment by schnillerman** [comment]
See also here: https://github.com/paperless-ngx/paperless-ngx/discussions/2753#discussioncomment-11780702

**Comment by enieuwy** [comment]
Bravo @ctout76 !
I used my existing compose and versions, just adding your gutenberg commands and 'PAPERLESS_WEBSERVER_WORKERS: 2`
Thank you!

**Comment by Tiff565** [comment]
Hey you guys using this with Container Manager on Synology, is there even a .yaml? I can only find .config for every separate container.

**Comment by ctout76** [comment]
Hi,

I'll soon be posting an update to the "compose.yaml" package so that it's up to date with Paperless-ngx v2.17.1 - Gotenberg 8.21.1 - Redis 8.0.3 - Postgres 17.5 (with possible conversion from earlier versions).

This one works but there are some errors since the latest versions of Gotenberg and Postgres in particular.

I'm finalizing the tests.

See you soon !



EDIT 1:
Below is a working version adapted for the latest versions !   :-)

Don't forget to personalize: UID and GID - TZ - IP adress

In the DSM configuration: Enable Reverse Proxy + Websocket, - Enable HTTP Compression - Enable port reuse to improve the capacity of simultaneous connections for the web service

Reboot the machine


https://hub.docker.com/r/pgautoupgrade/pgautoupgrade
is a PostgreSQL Docker container that automatically upgrades your database

For upgrade v16 to v17 (for example):
WARNING! Backup your data first!
Replace 

db:
`image: postgres:17.5`  by  `image: pgautoupgrade/pgautoupgrade:latest`

Once the upgrade is done, you can fix it again `image: postgres:17.5`


EDIT 2:
Redis 8.2.0 Update

**Comment by ctout76** [comment]
```
version: "3.9"
services:
  redis:
    image: redis:8.2.0
    command:
      - /bin/sh
      - -c
      - redis-server --requirepass redispass
    container_name: PaperlessNGX-REDIS
    hostname: paper-redis
    mem_limit: 512m
    mem_reservation: 256m
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    read_only: true
    user: 1026:100
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping || exit 1"]
    volumes:
      - /volume1/docker/paperlessngx/redis:/data:rw
    environment:
      TZ: Europe/Paris
    restart: unless-stopped

  db:
    image: postgres:17.5
    container_name: PaperlessNGX-DB
    hostname: paper-db
    mem_limit: 2g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "paperless", "-U", "paperlessuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/paperlessngx/db:/var/lib/postgresql/data:rw
    command: -c 'max_connections=1500'
    environment:
      POSTGRES_DB: paperless
      POSTGRES_USER: paperlessuser
      POSTGRES_PASSWORD: paperlesspass
    restart: unless-stopped

  gotenberg:
    image: gotenberg/gotenberg:8.21.1
    container_name: PaperlessNGX-GOTENBERG
    hostname: gotenberg
    security_opt:
      - no-new-privileges:true
    user: 1026:100
    command:
      - "gotenberg"
      - "--chromium-disable-javascript=true"
      - "--chromium-disable-routes=false"
      - "--chromium-allow-list=file:///tmp/.*"
      - "--api-timeout=300s"
      - "--libreoffice-start-timeout=30s"
      - "--libreoffice-auto-start=true"
      - "--libreoffice-restart-after=0"
    restart: unless-stopped

  tika:
    image: ghcr.io/paperless-ngx/tika:latest
    container_name: PaperlessNGX-TIKA
    hostname: tika
    security_opt:
      - no-new-privileges:true
    user: 1026:100
    restart: unless-stopped

  paperless:
    image: ghcr.io/paperless-ngx/paperless-ngx:2.17.1
    container_name: PaperlessNGX
    hostname: paperless-ngx
    mem_limit: 4g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "curl", "-fs", "-S", "--max-time", "2", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    ports:
      - 8777:8000
    volumes:
      - /volume1/docker/paperlessngx/data:/usr/src/paperless/data:rw
      - /volume1/docker/paperlessngx/media:/usr/src/paperless/media:rw
      - /volume1/docker/paperlessngx/export:/usr/src/paperless/export:rw
      - /volume1/docker/paperlessngx/consume:/usr/src/paperless/consume:rw
    environment:
      PAPERLESS_REDIS: redis://:redispass@paper-redis:6379
      PAPERLESS_DBENGINE: postgresql
      PAPERLESS_DBHOST: paper-db
      PAPERLESS_DBNAME: paperless
      PAPERLESS_DBUSER: paperlessuser
      PAPERLESS_DBPASS: paperlesspass
      PAPERLESS_WEBSERVER_WORKERS: 2
      PAPERLESS_TASK_WORKERS: 2
      PAPERLESS_WORKER_TIMEOUT: 9000
      USERMAP_UID: 1026
      USERMAP_GID: 100
      PAPERLESS_ENABLE_COMPRESSION: 0
      PAPERLESS_TIME_ZONE: Europe/Paris
      PAPERLESS_ADMIN_USER: admin
      PAPERLESS_ADMIN_PASSWORD: Paperlessadminpass
      PAPERLESS_URL: https://XXX.XXX.X.XXX:XXXX
      PAPERLESS_CSRF_TRUSTED_ORIGINS: https://XXX.XXX.X.XXX:XXXX
      PAPERLESS_OCR_LANGUAGE: fra
      PAPERLESS_OCR_USER_ARGS: '{"invalidate_digital_signatures": true}'
      PAPERLESS_CONSUMER_DISABLE: 1
      PAPERLESS_TIKA_ENABLED: 1
      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
      PAPERLESS_TIKA_ENDPOINT: http://tika:9998
    restart: unless-stopped
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_healthy
      tika:
        condition: service_started
      gotenberg:
        condition: service_started
```

**Comment by ctout76** [comment]
Update for P**ostgreSQL 18.1** (ONLY FOR NEW INSTALLATION - Databases lower than PostgreSQL 18.0 are not directly compatible, conversion is required)

Gotenberg 8.25.1

```
version: "3.9"
services:
  redis:
    image: redis:8.4.0
    command:
      - /bin/sh
      - -c
      - redis-server --requirepass redispass
    container_name: PaperlessNGX-REDIS
    hostname: paper-redis
    mem_limit: 512m
    mem_reservation: 256m
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    read_only: true
    user: 1026:100
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping || exit 1"]
    volumes:
      - /volume1/docker/paperlessngx/redis:/data:rw
    environment:
      TZ: Europe/Paris
    restart: unless-stopped

  db:
    image: postgres:18.1
    container_name: PaperlessNGX-DB
    hostname: paper-db
    mem_limit: 2g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "paperless", "-U", "paperlessuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/paperlessngx/db:/var/lib/postgresql:rw
    command: -c 'max_connections=10000'
    environment:
      POSTGRES_DB: paperless
      POSTGRES_USER: paperlessuser
      POSTGRES_PASSWORD: paperlesspass
      PGDATA: /var/lib/postgresql/18/docker
    restart: unless-stopped

  gotenberg:
    image: gotenberg/gotenberg:8.25.1
    container_name: PaperlessNGX-GOTENBERG
    hostname: gotenberg
    mem_limit: 4g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    user: 1026:100
    command:
      - "gotenberg"
      - "--chromium-disable-javascript=true"
      - "--chromium-disable-routes=false"
      - "--chromium-allow-list=file:///tmp/.*"
      - "--api-timeout=300s"
      - "--libreoffice-start-timeout=30s"
      - "--libreoffice-auto-start=true"
      - "--libreoffice-restart-after=0"
    restart: unless-stopped

  tika:
    image: docker.io/apache/tika:latest
    container_name: PaperlessNGX-TIKA
    hostname: tika
    security_opt:
      - no-new-privileges:true
    user: 1026:100
    restart: unless-stopped

  paperless:
    image: ghcr.io/paperless-ngx/paperless-ngx:2.20.3
    container_name: PaperlessNGX
    hostname: paperless-ngx
    mem_limit: 4g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "curl", "-fs", "-S", "--max-time", "2", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    ports:
      - 8777:8000
    volumes:
      - /volume1/docker/paperlessngx/data:/usr/src/paperless/data:rw
      - /volume1/docker/paperlessngx/media:/usr/src/paperless/media:rw
      - /volume1/docker/paperlessngx/export:/usr/src/paperless/export:rw
      - /volume1/docker/paperlessngx/consume:/usr/src/paperless/consume:rw
    environment:
      PAPERLESS_REDIS: redis://:redispass@paper-redis:6379
      PAPERLESS_DBENGINE: postgresql
      PAPERLESS_DBHOST: paper-db
      PAPERLESS_DBNAME: paperless
      PAPERLESS_DBUSER: paperlessuser
      PAPERLESS_DBPASS: paperlesspass
      PAPERLESS_WEBSERVER_WORKERS: 2
      PAPERLESS_TASK_WORKERS: 2
      PAPERLESS_WORKER_TIMEOUT: 9000
      USERMAP_UID: 1026
      USERMAP_GID: 100
      PAPERLESS_ENABLE_COMPRESSION: 0
      PAPERLESS_TIME_ZONE: Europe/Paris
      PAPERLESS_ADMIN_USER: admin
      PAPERLESS_ADMIN_PASSWORD: Paperlessadminpass
      PAPERLESS_URL: https://XXX.XXX.X.XXX:XXXX
      PAPERLESS_CSRF_TRUSTED_ORIGINS: https://XXX.XXX.X.XXX:XXXX
      PAPERLESS_OCR_LANGUAGE: fra
      PAPERLESS_OCR_USER_ARGS: '{"invalidate_digital_signatures": true}'
      PAPERLESS_CONSUMER_DISABLE: 1
      PAPERLESS_TIKA_ENABLED: 1
      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
      PAPERLESS_TIKA_ENDPOINT: http://tika:9998
    restart: unless-stopped
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_healthy
      tika:
        condition: service_started
      gotenberg:
        condition: service_started
```
